scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,442 @@
+    image.
+        cPD : contrast proton density
+            1 : reference proton density (MT-off)
+            2 : mean of positive and negative proton density (MT-on)
+        cT1 : contrast T1-weighted
+        num : numberator
+        den : denumerator
+
+    see Helms et al., 2008
+    https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.21732
+
+    Parameters
+    ----------
+    contrast_maps:      List of 3D-array constrats matrices : list of all
+                        contrast maps computed with compute_ihMT_contrasts
+    acq_parameters:     List of TR and Flipangle for ihMT and T1w images
+                        [TR, Flipangle]
+    Returns
+    -------
+    MT ratio and MT saturation matrice in 3D-array.
+    """
+    # Compute MT Ratio map
+    MTR = 100*((contrast_maps[2] -
+               (contrast_maps[4] + contrast_maps[3])/2)/contrast_maps[2])
+
+    # Compute MT saturation maps
+    cPD1 = contrast_maps[2]
+    cPD2 = (contrast_maps[4] + contrast_maps[3])/2
+    cT1 = contrast_maps[5]
+
+    Aapp_num = ((2*acq_parameters[0][0]/(acq_parameters[0][1]**2)) -
+                (2*acq_parameters[1][0]/(acq_parameters[1][1]**2)))
+    Aapp_den = (((2*acq_parameters[0][0])/(acq_parameters[0][1]*cPD1)) -
+                ((2*acq_parameters[1][0])/(acq_parameters[1][1]*cT1)))
+    Aapp = Aapp_num/Aapp_den
+
+    R1app_num = ((cPD1/acq_parameters[0][1]) - (cT1/acq_parameters[1][1]))
+    R1app_den = ((cT1*acq_parameters[1][1])/(2*acq_parameters[1][0]) -
+                 (cPD1*acq_parameters[0][1])/(2*acq_parameters[0][0]))
+    R1app = R1app_num/R1app_den
+
+    MTsat = 100*(((Aapp*acq_parameters[0][1]*acq_parameters[0][0]/R1app)/cPD2)-
+                 (acq_parameters[0][0]/R1app) - (acq_parameters[0][1]**2)/2)
+
+    return MTR, MTsat
+
+
+def threshold_ihMT_maps(computed_map, contrast_maps, in_mask,
+                        lower_threshold, upper_threshold, idx_contrast_list):
+    """
+    Remove NaN and apply different threshold based on
+       - maximum and minimum threshold value
+       - T1 mask
+       - combination of specific contrasts maps
+
+    Parameters
+    ----------
+    computed_map        3D-Array data.
+                        Myelin map (ihMT or non-ihMT maps)
+    contrast_maps       List of 3D-Array. File must containing the
+                        6 contrasts maps.
+    in_mask             Path to binary T1 mask from T1 segmentation.
+                        Must be the sum of GM+WM+CSF.
+    lower_threshold     Value for low thresold <int>
+    upper_thresold      Value for up thresold <int>
+    idx_contrast_list   List of indexes of contrast maps corresponding to
+                        that of input contrast_maps ex.: [0, 2, 5]
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded matrix in 3D-array.
+    """
+    # Remove NaN and apply thresold based on lower and upper value
```

Comment:
If we remove NaN, why is there a piece of code to discard NaN warning/error ?

Reply:
I have this error for the maps computing before threshold (functions compute_*_maps). This error is not only due to the presence of NaN, I modified the comment.